### ISSUE TITLE
Bug: setState doesn't update correctly when using updater functions

### ISSUE THREAD

**Issue by eddhurst** [mr-hooks/use-query-parameter#11]
Issue:
setState, when passed with an updater function to calculate the next state from the pending state, doesn't update correctly. 

Note - query parameter turns everything into a string, so using parseInt here to convert back to numbers.

```jsx
const [count, setCount] = useQueryParameter('count', 4);

const handleClick = () => {
  setCount(5)   // works fine
  setCount(parseInt(count, 10) + 1) // works fine
  setCount((count) => parseInt(count, 10) + 1) // does not resolve the updater function passed in
}
```

**Comment by code-forger** [comment]
Hi @eddhurst are you interested in hunting this bug yourself? if so can I assign this issue to you?

**Comment by eddhurst** [comment]
100% that'd be great. I'll raise a PR shortly

**Comment by code-forger** [comment]
Landed in [v0.0.4](https://www.npmjs.com/package/@mr-hooks/use-query-parameter/v/0.0.4)
